### PR TITLE
add pow

### PR DESCRIPTION
**Pull request by alanvardy**
Ok, I am now a fan of Pow.

Documentation is good, and it even tells you how to set it up in the command line when you run the commands.

Oh, and I wrote my first little plug!

### CODE REVIEW

File: lib/health_web/router.ex [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
+      error_handler: Pow.Phoenix.PlugErrorHandler
+    end
+
+    pipeline :browser do
+      plug :accepts, ["html"]
+      plug :fetch_session
+      plug :fetch_flash
+      plug :protect_from_forgery
+      plug :put_secure_browser_headers
+      plug Plugs.CurrentUser
```
Comment (alanvardy) [inline lib/health_web/router.ex]: There's my plug!

File: lib/health_web/router.ex [hunk 2]
```diff
@@ -1,22 +1,39 @@
...
-    get "/", PageController, :index
+    get "/", HealthWeb.PageController, :index
+    pow_routes()
+  end
+
+  scope "/", HealthWeb do
+    pipe_through [:browser, :protected]
+    # Add your protected routes here
+    get "/protected", PageController, :protected
   end
```
Comment (alanvardy) [inline lib/health_web/router.ex]: I like this style of having all protected routes in one block, makes it really obvious what a not-logged in person can access.

File: lib/health_web/templates/layout/app.html.eex [hunk 3]
```diff
@@ -11,7 +11,7 @@
   <body>
     <%= render "_impersonate.html", conn: @conn %>
     <%= render "_staging.html", conn: @conn %>
-    <%= render "_navbar.html", conn: @conn %>
+    <%= render "_navbar.html", conn: @conn, current_user: @current_user %>
```
Comment (alanvardy) [inline lib/health_web/templates/layout/app.html.eex]: These module attributes have to be explicitly passed to partials

File: lib/health_web/plugs/current_user.ex [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    Adds current_user to assigns if user is logged in
+    (for use in templates)
+    """
+
+  def init(options), do: options
+
+  def call(conn, _opts) do
+    Plug.Conn.assign(conn, :current_user, Pow.Plug.current_user(conn))
+  end
+end
```
Comment (alanvardy) [inline lib/health_web/plugs/current_user.ex]: It ain't much to looks at, but sure as hell beats calling `Pow.Plug.current_user(@conn)` in the template every time!

File: lib/health_web/plugs/current_user.ex [hunk 5]
```diff
@@ -0,0 +1,12 @@
+defmodule HealthWeb.Plugs.CurrentUser do
+  @moduledoc """
+    Adds current_user to assigns if user is logged in
+    (for use in templates)
+    """
```
Comment (alanvardy) [inline lib/health_web/plugs/current_user.ex]: I just found out that these block "heredoc" comments support markdown!

File: lib/health_web/plugs/current_user.ex (hunk 4 above)
Comment (danschultzer) [reply lib/health_web/plugs/current_user.ex]: Hey man, just searched for Pow users in github to see what people are building. `Pow.Plug.Session` already sets a `:current_user` assigns already so no need for the plug or to call `Pow.Plug.current_user/1`, @current_user is already available :)

It's documented in the readme here, but I should add it to the `current_user/1` docs as well: https://hexdocs.pm/pow/README.html#current-user-and-sign-out-link

Comment (alanvardy) [reply lib/health_web/plugs/current_user.ex]: Awesome, thanks for checking this out!

I'll have to do a refactor :)

**Pull request merged**